apps/admin-gui/src/app/shared/components/dialogs/edit-application-form-item-dialog/edit-application-form-item-dialog.component.ts

```diff
@@ -41,7 +41,7 @@ export class EditApplicationFormItemDialogComponent implements OnInit {
   hiddenValues: HiddenEnum[] = ['NEVER', 'ALWAYS', 'IF_EMPTY', 'IF_PREFILLED'];
   disabledValues: DisabledEnum[] = ['NEVER', 'ALWAYS', 'IF_EMPTY', 'IF_PREFILLED'];
   possibleDependencyItems: ApplicationFormItem[] = [];
-  htmlInput = new FormControl('', [this.escapeService.htmlContentValidator()]);
+  inputFormGroup: FormGroup = null;
```

Comment:
This is a typed form group so we should also define the type to get the benefits of type checking and safety. But since the number and name of controls is generated dynamically based on the form item type and supported languages we can't use exact type. The `Record<string, FormControl<string>>` says it will be some structure with key of type `string` and value of type `FormControl<string>`. With this change you can remove any type casting mainly in the `submit()` method.
```suggestion
  inputFormGroup: FormGroup<Record<string, FormControl<string>>> = null;
```